src/main/java/org/sunbird/ratings/service/RatingServiceImpl.java

```diff
@@ -0,0 +1,174 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import org.sunbird.cassandra.utils.CassandraOperation;
+import org.sunbird.common.model.SBApiResponse;
+import org.sunbird.common.util.Constants;
+import org.sunbird.core.producer.Producer;
+import org.sunbird.ratings.exception.ValidateRatingException;
+import org.sunbird.ratings.model.RatingMessage;
+import org.sunbird.ratings.model.RequestRating;
+import org.sunbird.ratings.model.UpdatedValues;
+
+import java.util.*;
+import java.util.regex.Pattern;
+
+@Service
+public class RatingServiceImpl implements RatingService {
+    private ObjectMapper mapper = new ObjectMapper();
+
+    private Logger logger = LoggerFactory.getLogger(getClass().getName());
+
+    @Autowired
+    CassandraOperation cassandraOperation;
+
+    @Autowired
+    Producer producer;
+
+    //TO DO
+    @Override
+    public SBApiResponse getRatings(String activity_Id, String activity_Type, String userId) {
+        return null;
+    }
+
+    //TO DO
+    @Override
+    public SBApiResponse getUsers(String activity_Id, String activity_Type, String userId) {
+        return null;
+    }
+
+    //TO DO
+    @Override
+    public SBApiResponse upsertRating(String activity_Id, String activity_Type, String userId) {
+        return null;
+    }
+
+    //TO DO
+    @Override
+    public SBApiResponse getRatingSummary(String activity_Id, String activity_Type) {
+        return null;
+    }
+
+    @Override
+    public SBApiResponse upsert(RequestRating requestRating) {
+        UUID timeBasedUuid = UUIDs.timeBased();
+
+        SBApiResponse response = new SBApiResponse(Constants.API_RATINGS_ADD);
+        RatingMessage msg = new RatingMessage();
+
+        try {
+            validateRatingsInfo(requestRating);
+
+            Map<String, Object> request = new HashMap<>();
+            request.put(Constants.ACTIVITY_ID, requestRating.getActivity_Id());
+            request.put(Constants.ACTIVITY_TYPE, requestRating.getActivity_type());
+            request.put(Constants.RATINGS_USER_ID, requestRating.getUserId());
+
+
+            List<Map<String, Object>> existingDataList = cassandraOperation.getRecordsByProperties(Constants.KEYSPACE_SUNBIRD,
+                    Constants.TABLE_RATINGS, request, null);
+
+            if (!existingDataList.isEmpty()) {
+
+                Map<String, Object> updateRequest = new HashMap<>();
+                updateRequest.put(Constants.RATING, requestRating.getRating());
+                updateRequest.put(Constants.REVIEW, requestRating.getReview());
+                updateRequest.put(Constants.UPDATED_ON, timeBasedUuid);
+
+                Map<String, Object> prevInfo = existingDataList.get(0);
+                cassandraOperation.updateRecord(Constants.KEYSPACE_SUNBIRD, Constants.TABLE_RATINGS, updateRequest,
+                        request);
+
+                msg.setAction("ratingUpdate");
```

Comment:
We should have a constructor that takes in the basic types so that we can avoid so many setter calls

Reply:
done